common/src/main/java/io/druid/math/expr/Function.java

```diff
@@ -242,6 +243,27 @@ protected ExprEval eval(double param)
     }
   }
 
+  class Div extends DoubleParamMath
+  {
+    @Override
+    public String name()
+    {
+      return "div";
+    }
+
+    @Override
+    protected ExprEval eval(final long x, final long y)
+    {
+      return ExprEval.of(x / y);
+    }
+
+    @Override
+    protected ExprEval eval(final double x, final double y)
+    {
+      return ExprEval.of((long) (x / y));
```

Comment:
I think `(long) (x / y)` makes more sense to me. That way the division is done on the numbers in their 'native' form (which might be outside the range of `long`) and then the result is converted to the nearest integer form.